### PR TITLE
Bug 1274679 - Use server time for end time in metrics requests

### PR DESCRIPTION
**Pull request by spadgett**
Don't specify an `end` query parameter on metrics requests so that the server's current time is used as an end time. Show the server error message in the alert details if available.

https://bugzilla.redhat.com/show_bug.cgi?id=1274679

### CODE REVIEW

**Comment by spadgett** [comment]
@jwforres Please take a look. The `Date` header is not in the list that comes back from `$http`, so I'm falling back to showing the error message from Hawkular. It might be because `Date` isn't in the list of CORS allowed headers?

**Comment by jwforres** [comment]
@spadgett yep
```
allowedHeaders = []string{"Content-Type", "Content-Length", "Accept-Encoding", "X-CSRF-Token", "Authorization", "X-Requested-With", "If-Modified-Since"}
```

@smarterclayton @liggitt we've hit a case where we need a header back from the API server that isn't already whitelisted by the code

**Comment by spadgett** [comment]
This is actually a hawkular-metrics header, although we might want `Date` to someday to fix #2568 ... maybe?

**Comment by jwforres** [comment]
@spadgett ah right so its up to the hawkular api to add that to their list of allowed headers

**Comment by spadgett** [comment]
@jwforres Right, but we might raise this as a general issue with them. There isn't a good way to deal with clock skew that I've found using their REST API.

**Comment by liggitt** [comment]
I can't think of a good reason not to whitelist the Date header in our code, and ask them to do the same. Client should use the header if available

File: assets/app/scripts/directives/metrics.js [hunk 1]
```diff
@@ -208,32 +208,44 @@ angular.module('openshiftConsole')
...
+                var alert = {
                   type: "error",
                   message: "Error fetching " + metric.id + " for container " + container.name + "."
                 };
-              }).finally(function() {
+
+                if (response.data && response.data.errorMsg) {
+                  alert.details = response.data.errorMsg;
+                } else if (response.status === 0) {
+                  alert.details = "Could not connect to metrics service. Make sure you have a network connection and the metrics service is running.";
```
Comment (jwforres) [inline assets/app/scripts/directives/metrics.js]: I would just leave it at "Could not connect to the metrics service"

**Comment by jwforres** [comment]
one nit, otherwise LGTM

**Comment by spadgett** [comment]
@jwforres thanks, updated the message.

**Comment by jwforres** [comment]
[merge]

**Comment by spadgett** [comment]
Unrelated error, [merge] again.

```
!!! Error in test/cmd/newapp.sh:25
	'oc new-app php mysql -l no-source=php-mysql' exited with status 1
Call stack:
	1: test/cmd/newapp.sh:25 main(...)
Exiting with status 1
!!! Error in hack/test-cmd.sh:276
	'${test}' exited with status 1
Call stack:
	1: hack/test-cmd.sh:276 main(...)
Exiting with status 1
[FAIL] !!!!! Test Failed !!!!
```

**Comment by jwforres** [comment]
unrelated failure
```
--- FAIL: TestTriggers_imageChange-2 (6.15s)
	deploy_trigger_test.go:164: unexpected error: imageStream "test-image-stream" cannot be updated: the object has been modified; please apply your changes to the latest version and try again
FAIL
```

[merge]

**Comment by smarterclayton** [comment]
Spawned https://github.com/openshift/origin/issues/5421

**Pull request merged**